File: py/medium/2870/2870.py
from collections import Counter
from typing import List


class Solution:
    def minOperations(self, nums: List[int]) -> int:
        counter = Counter(nums)
        if 1 in counter.values():
            return -1
        answer = 0
        for value in counter.values():
            answer += value // 3 + (value % 3 != 0)
        return answer


solution = Solution()

# assert solution.minOperations([2, 3, 3, 2, 2, 4, 2, 3, 4]) == 4
# assert solution.minOperations([2, 1, 2, 2, 3, 3]) == -1
assert solution.minOperations([16, 16, 16, 19, 16, 3, 16, 8, 16, 16, 16, 19, 3, 16, 16]) == -1

File: elixir/medium/2870/2870.exs
defmodule Solution do
  @spec min_operations(nums :: [integer]) :: integer
  def min_operations(nums) do
    nums
    |> Enum.frequencies()
    |> IO.inspect()
    |> Map.values()
    |> IO.inspect()
  end

  def boolean_to_integer(bool) do
    if bool, do: 1, else: 0
  end
end

Solution.min_operations([2, 3, 3, 2, 2, 4, 2, 3, 4])

# ExUnit.start()

# defmodule AssertionTest do
#   use ExUnit.Case, async: true

#   test "test_1" do
#     assert Solution.min_operations([2, 3, 3, 2, 2, 4, 2, 3, 4]) == 4
#   end

#   test "test_2" do
#     assert Solution.min_operations([2, 1, 2, 2, 3, 3]) == -1
#   end
# end
